Bryanna/map-implementation

bry-gavino opened the pull request:
_Describe your feature here_

Added a link at the top of the screen which is supposed to feature a display of a map from my Mapbox API, using my personal access token. The map won't display, but the link works.

_Optional - any related PRs you're waiting on, or PRs that will conflict, etc_

N/A

_Optional - if you added anything to the database through migration(s)_

<img width="316" alt="Screen Shot 2020-10-04 at 8 28 40 PM" src="https://user-images.githubusercontent.com/60284767/95037542-43529c00-0680-11eb-9ba9-d6e043786f6c.png">


_Required for visual changes_

CC: @kwfk

bry-gavino committed:
feat(map): mapbox Test

implemented mapbox api but struggling trying to format and stylize correctly to have it appear on
the screen

bry-gavino committed:
feat(map): mapbox test

implement mapbox api to homepage

bry-gavino committed:
feat: mapbox Map Visual

implemeted a map visual of mapbox into homepage

kwfk commented on pages/map.tsx:
```diff
@@ -0,0 +1,28 @@
+import Layout from 'components/Layout';
```
delete this file

kwfk commented on styles/mapbox-gl.css:
```diff
@@ -0,0 +1,692 @@
+.mapboxgl-map {
```
delete this file

kwfk commented on pages/index.tsx:
```diff
@@ -1,7 +1,12 @@
 import { Button, CircularProgress } from '@material-ui/core';
 import styles from '../styles/Home.module.css';
 
+const DynamicComponentWithNoSSR = dynamic(() => import('../components/Map'), {
```
change name to something more descriptive

kwfk commented on pages/index.html:
```diff
@@ -0,0 +1,40 @@
+<script src='https://api.mapbox.com/mapbox-gl-js/v1.12.0/mapbox-gl.js'></script>
```
delete this file

kwfk commented on package.json:
```diff
@@ -34,16 +34,19 @@
     "db-migrate": "^0.11.11",
     "db-migrate-base": "^2.3.0",
     "db-migrate-pg": "^1.2.2",
+    "mapbox-gl": "^1.12.0",
```
remove this package

kwfk commented on package-lock.json:
```diff
@@ -0,0 +1,297 @@
+{
```
remove this file

kwfk commented on components/Map.tsx:
```diff
@@ -0,0 +1,22 @@
+  return (
+    <ReactMapGL
+      mapStyle="mapbox://styles/mapbox/streets-v9"
+      mapboxApiAccessToken="pk.eyJ1IjoiYnJ5LWdhdmlubyIsImEiOiJja2ZydDZjdHkwOXc1MnJxcWJqNHRsejF3In0.wUyQfr7dEqtTx8Q4r2EYiA"
```
generally, we don't want API keys to be hard-coded onto our application
move this into an environment variable

add a new key to `.env.local.development` with the prefix `NEXT_PUBLIC_` (this allows it to be accessed from the browser where this component will be rendered) so for example `NEXT_PUBLIC_MAPBOX_API_KEY`
add the same entry to `.env.example` but without the actual key so that others know what they should put in their env file

kwfk reviewed (changes requested):
some small changes needed but nice work on first PR!

kwfk commented:
also change the PR title to what this feature is

bry-gavino committed:
docs: pr changes

deleted a few unnecessary files, removed unused imports, fixed the hard-coding of the API

kwfk commented on prisma/.env.example:
```diff
@@ -4,4 +4,5 @@
-DATABASE_URL="postgresql://postgres@localhost:5432/nbjc_development?schema=public"
\ No newline at end of file
+DATABASE_URL="postgresql://postgres@localhost:5432/nbjc_development?schema=public"
+NEXT_PUBLIC_MAPBOX_API_KEY=mapboxapikey
```
we don't need the mapbox api key here

kwfk reviewed (approved):
make one small change to `prisma/.env.example` then you're free to merge (make sure to **squash** and merge)!
nice work!

bry-gavino committed:
docs: removed unnecessary api key assignment
